### PR TITLE
persist: add a delete function to Blob

### PR DESCRIPTION
**Pull request by ruchirK**

### CODE REVIEW

**Comment by ruchirK** [comment]
RFAL! I decided to break this up into two prs so that we could hash out the api / testing of the lower level blob api here and then the usage of it inside Indexed in the next one.

My two open questions for this one were:

- is returning `Result<(), Error>` the right api, or should this do something more like `Result<bool, ()>` where it returns true if the key was actually deleted, and false if it didn't exist in the map, and error if there was some other error
- is it fine to not do the directory fsync thing for deletes? (i think so because we just did a similar thing for atomic writes)

File: src/persist/src/storage.rs [hunk 1]
```diff
@@ -74,6 +74,9 @@ pub trait Blob {
     /// Inserts a key-value pair into the map.
     fn set(&mut self, key: &str, value: Vec<u8>, allow_overwrite: bool) -> Result<(), Error>;
 
+    /// Remove a key from the map.
```
Comment (danhhz) [inline src/persist/src/storage.rs:77]: Document that is errors if the key does not exist

**Review by danhhz — APPROVED** [review]
LGTM!

> is returning Result<(), Error> the right api, or should this do something more like Result<bool, ()> where it returns true if the key was actually deleted, and false if it didn't exist in the map, and error if there was some other error

This interface seems correct to me. Do we need to distinguish these cases yet for the upcoming work or is this just musing?

> is it fine to not do the directory fsync thing for deletes? (i think so because we just did a similar thing for atomic writes)

Fine for now for sure. This one is even less of an issue than the atomic write since it just leaks the deleted file

**Comment by ruchirK** [comment]
> Do we need to distinguish these cases yet for the upcoming work or is this just musing?

just musing

**Comment by ruchirK** [comment]
tftr!

**Pull request merged**